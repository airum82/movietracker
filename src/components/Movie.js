import React from 'react';
import './movie.css';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { promptCreate } from '../actions';
import { postFavorite } from '../ApiCall';
import { toggleFavorite } from '../actions';
import { removeFavorite } from '../ApiCall';

export const Movie = (props) => {
  
  const { 
    title, 
    overview, 
    poster, 
    ratings, 
    id, 
    checkUser,
    favorite 
  } = props;

  const handleFavorite = (props) => {

    if(!props.user.name) {
      props.handlePromptCreate()
    } else if(favorite) {
      removeFavorite(id, props.user.id)
    } else {
      postFavorite(props, props.user.id);
    }
  }

  return (
    <div className='movie'>
      <div className='title-ratings-div'>
        <h1>{title}</h1>
        <h4>{ratings}</h4>
      </div>
      <div className="poster-div" >
        <img className='poster' src={`https://image.tmdb.org/t/p/w600_and_h900_bestv2/${poster}`} />
      </div>
      <div className="overview">
        <p>{overview}</p>
      </div>
      <div className="add-favorite-div">
      <button
        className="add-favorite-button"
        onClick={() => handleFavorite(props)}
        value={id}
      >favorite</button>
      </div>
    </div>
  )
}

const mapStateToProps = (state) => ({
  nowPlaying: state.nowPlaying,
  user: state.user
});

const mapDispatchToProps = (dispatch) => ({
  handlePromptCreate : () => dispatch(promptCreate())
})


export default connect(mapStateToProps, mapDispatchToProps)(Movie);

Movie.Proptypes = {
  title: PropTypes.string,
  overview: PropTypes.string,
  poster: PropTypes.string,
  ratings: PropTypes.string
}