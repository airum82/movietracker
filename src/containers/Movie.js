import React from 'react';
import './movie.css';
import PropTypes from 'prop-types';

export const Movie = (props) => {
  
  const { 
    title, 
    overview, 
    poster, 
    ratings,
    id,
    handleFavorite
  } = props;

  const toggleFavorite = (event) => {
    const id = event.target.value;
    console.log(id)
    handleFavorite(id)
  }

  return (
    <div className='movie'>
      <div className='title-ratings-div'>
        <h1 className="movie-title">{title}</h1>
        <h4>{ratings}</h4>
      </div>
      <div className="poster-div" >
        <img 
          className='poster' 
          src={`https://image.tmdb.org/t/p/w600_and_h900_bestv2/${poster}`} 
        />
      </div>
      <div className="overview">
        <p>{overview}</p>
      </div>
      <div className="add-favorite-div">
        <button
          className="add-favorite-button"
          onClick={toggleFavorite}
          value={id}
        >favorite</button>
      </div>
    </div>
  );
};

Movie.Proptypes = {
  title: PropTypes.string,
  overview: PropTypes.string,
  poster: PropTypes.string,
  ratings: PropTypes.string
};