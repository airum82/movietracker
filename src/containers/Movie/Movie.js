import React from 'react'
import { connect } from 'react-redux';

export const Movie = (nowPlaying) => {
  
  const {title, overview, poster, ratings} = nowPlaying

  return (
    <div className='movie'>
      <h1>{title}</h1>
      <p>{overview}</p>
      <img src={`https://image.tmdb.org/t/p/w600_and_h900_bestv2/${poster}`} />
      <h4>{ratings}</h4>
    </div>
  )
}

const mapStatetoProps = (state) => ({
  movies: state.movies
})

export default connect(mapStatetoProps, null)(Movie);