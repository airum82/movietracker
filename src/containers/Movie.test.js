import React, { Component } from 'react';
import Movie from './Movie';
import { shallow, mount } from 'enzyme';

describe('Movie', () => {

  it('should match snapshot', () => {
    const wrapper = shallow(<Movie />)

    expect(wrapper).toMatchSnapshot()
  })
})