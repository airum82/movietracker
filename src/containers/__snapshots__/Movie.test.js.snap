// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Movie />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="title-ratings-div"
>
          <h1 />
          <h4 />
</div>,
        <div
          className="poster-div"
>
          <img
                    className="poster"
                    src="https://image.tmdb.org/t/p/w600_and_h900_bestv2/undefined"
          />
</div>,
        <div
          className="overview"
>
          <p />
</div>,
      ],
      "className": "movie",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1 />,
            <h4 />,
          ],
          "className": "title-ratings-div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "h4",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            className="poster"
            src="https://image.tmdb.org/t/p/w600_and_h900_bestv2/undefined"
/>,
          "className": "poster-div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "poster",
            "src": "https://image.tmdb.org/t/p/w600_and_h900_bestv2/undefined",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p />,
          "className": "overview",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="title-ratings-div"
>
            <h1 />
            <h4 />
</div>,
          <div
            className="poster-div"
>
            <img
                        className="poster"
                        src="https://image.tmdb.org/t/p/w600_and_h900_bestv2/undefined"
            />
</div>,
          <div
            className="overview"
>
            <p />
</div>,
        ],
        "className": "movie",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1 />,
              <h4 />,
            ],
            "className": "title-ratings-div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "h4",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              className="poster"
              src="https://image.tmdb.org/t/p/w600_and_h900_bestv2/undefined"
/>,
            "className": "poster-div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "poster",
              "src": "https://image.tmdb.org/t/p/w600_and_h900_bestv2/undefined",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p />,
            "className": "overview",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
